pyvista/plotting/plotting.py

```diff
@@ -3715,6 +3721,15 @@ def add_volume(
         elif not isinstance(scalars, np.ndarray):
             scalars = np.asarray(scalars)
 
+        if scalars.ndim != 1:
+            if scalars.ndim != 2:
+                raise ValueError('`add_volume` only supports scalars with 1 or 2 dimensions')
+            if scalars.shape[1] != 4 or scalars.dtype != np.uint8:
+                raise ValueError(
```

Comment:
I'm half wondering if these should be TypeErrors. No time to look up precedents in the code base right now.

Reply:
We typically use `ValueErrors` for numpy types.